core/cloudflow-operator/src/test/scala/cloudflow/operator/runner/FlinkRunnerSpec.scala

```diff
@@ -78,6 +78,115 @@ class FlinkRunnerSpec extends WordSpecLike with OptionValues with MustMatchers w
       replicas = None
     )
 
+    "read values from pod configuration and put it in both " in {
+
+      val crd = FlinkRunner.resource(
+        deployment = deployment,
+        app = app,
+        configSecret = Secret(
+          metadata = ObjectMeta(),
+          data = Map(cloudflow.operator.event.ConfigInputChangeEvent.PodsConfigDataKey ->
+            """
+              | kubernetes.pods.pod.containers.container {
+              |       env = [
+              |          { name = "JAVA_OPTS"
+              |            value = "-XX:MaxRAMPercentage=40.0"
+              |          },{
+              |            name = "FOO"
+              |            value = "BAR"
+              |          }
+              |        ]
+              |        resources {
+              |            requests {
+              |              memory = "512M"
+              |            }
+              |            limits {
+              |              memory = "1024M"
+              |            }
+              |         }
+              |}
+              |        """.stripMargin.getBytes())
+        ),
+        namespace = namespace
+      )
+
+      crd.metadata.namespace mustBe namespace
+      crd.metadata.name mustBe appId
+      val imageWithoutRegistry = image.split("/").tail.mkString("/")
+      crd.spec.image must include(imageWithoutRegistry)
+      crd.spec.entryClass mustBe "cloudflow.runner.Runner"
+
+      crd.spec.volumes mustBe Vector(
+        Volume("config-map-vol", Volume.ConfigMapVolumeSource("configmap-some-app-id")),
+        Volume("persistent-storage-vol", Volume.PersistentVolumeClaimRef("some-app-id-pvc")),
+        Volume("secret-vol", Volume.Secret("flink-streamlet")),
+        Runner.DownwardApiVolume
+      )
+      crd.spec.jobManagerConfig.envConfig.get.env.get mustBe Vector(EnvVar("FOO",EnvVar.StringValue("BAR")))
+      crd.spec.taskManagerConfig.envConfig.get.env.get mustBe Vector(EnvVar("FOO",EnvVar.StringValue("BAR")))
+      crd.spec.jobManagerConfig.resources.get.requests mustBe Map(Resource.memory -> "512M")
+      crd.spec.taskManagerConfig.resources.get.requests mustBe Map(Resource.memory -> "512M")
+      crd.spec.jobManagerConfig.resources.get.limits mustBe Map(Resource.memory -> "1024M")
+      crd.spec.taskManagerConfig.resources.get.limits mustBe Map(Resource.memory -> "1024M")
+    }
+
+    "read values from pod configuration key JAVA_OPTS and " +
+      "put it in Flink conf in env.java.opts" in {
```

Comment:
no need to break line I think or is this scalafmt?

Reply:
my bad